engine/src/main/java/org/hibernate/search/query/dsl/impl/MoreLikeThisBuilder.java

```diff
@@ -188,18 +188,11 @@ private Query maybeExcludeComparedEntity(Query query) {
 		// but at this stage we could have documents reordered and thus with a different id
 		// Maybe a Filter would be more efficient?
 		if ( excludeEntityCompared && documentNumber != null ) {
-			BooleanQuery booleanQuery;
-			if ( ! ( query instanceof BooleanQuery ) ) {
-				booleanQuery = new BooleanQuery();
-				booleanQuery.add( query, BooleanClause.Occur.MUST );
-			}
-			else {
-				booleanQuery = (BooleanQuery) query;
-			}
-			booleanQuery.add(
-					new ConstantScoreQuery( findById ),
-					BooleanClause.Occur.MUST_NOT );
-			return booleanQuery;
+			return new BooleanQuery.Builder()
```

Comment:
@mincong-h, previously this would have re-used an incoming `BooleanQuery`, whereas now it always creates a new builder. Was there any specific reason for getting rid of this optimization?